### PR TITLE
feat: make CompactContractBytecodeCow implement Artifact

### PR DESCRIPTION
**Pull request by DaniPopes**

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm

File: src/compilers/mod.rs
Comment (mattsse) [inline src/compilers/mod.rs:1]: this will likely conflict with #128

@klkvr should we exclude these changes here?

Comment (DaniPopes) [reply src/compilers/mod.rs:1]: it doesnt

Comment (DaniPopes) [reply src/compilers/mod.rs:1]: ah nvm, it's just formatting changes

**Pull request merged**